src/compiler/checker.ts

```diff
@@ -3572,7 +3572,18 @@ module ts {
             return false;
         }
 
+        // Since removeSubtypes checks the subtype relation, and the subtype relation on a union
+        // may attempt to reduce a union, it is possible that removeSubtypes could be called
+        // recursively on the same set of types.

```

Comment:
For a future reader, this comment might leave them thinking "and...?"

Explain further that the stack is used to detect when subtype reduction is currentlytaking place for a given union type.